File: C-Sharp/11.cs
public class Solution {
    public int MaxArea(int[] height)
    {
        int medium = height.Length / 2;
        int max = 0;
        int pointerLeft = 0;
        int pointerRigth = height.Length - 1;

        while (pointerLeft != pointerRigth)
        {
            int minIndex = Math.Min(height[pointerLeft], height[pointerRigth]);
            
            int value = (pointerRigth - pointerLeft) * minIndex;

            if (value > max) max = value;

            if (height[pointerLeft] < height[pointerRigth])
            {
                pointerLeft++;
            }
            else
            {
                pointerRigth--;
            }
        }

        return max;
    }
}

File: Python/11.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        medium: int = int(len(height) / 2)
        max: int = 0
        pointerLeft = 0
        pointerRigth = (len(height)) - 1

        while pointerRigth != pointerLeft:
            min_index = min(height[pointerLeft], height[pointerRigth])
            value = (pointerRigth - pointerLeft) * min_index
            if value > max:
                max = value

            if height[pointerLeft] < height[pointerRigth]:
                pointerLeft += 1
            else:
                pointerRigth -= 1

        return max